shodan/client.py

```diff
@@ -294,13 +295,13 @@ def _request(self, function, params, service='shodan', method='get'):
         try:
             method = method.lower()
             if method == 'post':
-                data = self._session.post(base_url + function, params)
+                data = self._session.post(base_url + function, params, timeout=REQUESTS_TIMEOUT)
```

Comment:
I suggest to set the timeout on the session, that's easier (`self._session.timeout = 30` or similar)

Reply:
I agree. I thought about speficying the timeout directly when the session object is created but it's no supported and there's no setter for it. I didn't think about assigning directly to the attribute. I'll do that.

Reply:
Nevermind, it doesn't work. The session object does not have a timeout attribute at all. The timeout is directly passed to the method.

https://github.com/psf/requests/blob/09fd857eb1448a8234e39bc39e2c795d4c980bfb/requests/sessions.py#L337

Reply:
Uh, ok that seems to have worked previously, but not anymore.

Other solutions:
 * A shim on the request: https://github.com/psf/requests/issues/2011#issuecomment-490050252
 * in intelmq we have our own TimeoutHTTPAdapter https://github.com/certtools/intelmq/blob/develop/intelmq/lib/utils.py#L692-L726